src/main/java/communication/MessageIDEvaluator.java

```diff
@@ -0,0 +1,142 @@
+package communication;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+
+import control.FBase;
+import de.hasenburg.fbase.model.GetMissedMessageResponse;
+import exceptions.FBaseCommunicationException;
+import exceptions.FBaseStorageConnectorException;
+import model.data.MessageID;
+import model.data.NodeID;
+
+/**
+ * The {@link MessageIDEvaluator} manages all messageIDs received.
+ * 
+ * Everytime,
+ * {@link Subscriber#interpreteReceivedEnvelope(model.messages.Envelope, org.zeromq.ZMQ.Socket)}
+ * is called, the subscriber adds the ID of the contained envelope if one is set using
+ * {@link #addReceivedMessageID(MessageID)}.
+ * 
+ * The {@link MessageIDEvaluator} continuously runs through all not evaluated messageIDs. If
+ * he finds a gap, he will create a messageSender and ask the related node about the send
+ * data.
+ * 
+ * @author jonathanhasenburg
+ *
+ */
+public class MessageIDEvaluator {
```

Comment:
should be MessageIdEvaluator in correct camel case...

Reply:
I just checked our spelling in all other classes, and we did it like above. I think we should keep it like that to be consistent